src/main/react/src/views/Profile.js

```diff
@@ -0,0 +1,157 @@
+
+  const followTag = async (user, tagId) => {
+    if (!followedTagIds.includes(tagId)) {
+      setFollowedTagIds(followedTagIds.concat(tagId));
+      let idToken = await user.getIdToken();
+      fetch(requestRoute + "api/user?type=FOLLOW_TAG&tagID=" + tagId + "&token=" + idToken, { method: "PUT" });
+    }
+  };
+
+  const unfollowTag = async (user, tagId) => {
+    if (followedTagIds.includes(tagId)) {
+      setFollowedTagIds(followedTagIds.filter(id => id !== tagId));
+      let idToken = await user.getIdToken();
+      fetch(requestRoute + "api/user?type=UNFOLLOW_TAG&tagID=" + tagId + "&token=" + idToken, { method: "PUT" });
+    }
+  };
+
+  const getFollowedTags = async user => {
+    let idToken = await user.getIdToken();
+    let tagsJson = await fetch(requestRoute + "api/user?type=FOLLOW_TAG&token=" + idToken);
+    let tagsData = await tagsJson.json();
+    return tagsData;
+  };
+
+  let listener = app.auth().onAuthStateChanged(user => {
+    if (user) {
+      listener(); // unsubscribe from this listener
+      setSignedIn(true);
+      setProfilePic(user.providerData[0].photoURL);
+    }
+  });
+
+  useEffect(() => {
+    getTags().then(setAllTags);
+    let listener2 = app.auth().onAuthStateChanged(async user => getFollowedTags(user).then(setFollowedTagIds));
+    return () => {
+      listener2();
+      setFollowedTagIds([]);
+      setAllTags({});
+    };
+  }, []);
+
+  return (
+    <>
+      {signedIn && (
+        <>
+          <CCard>
+            <CCardBody>
+              <CRow>
+                <CCol sm={3}>
+                  <CImg
+                    src={profilePic}
+                    className="c-avatar-img"
+                    alt={app.auth().currentUser.email}
+                    style={{ width: "80%" }}
+                  />
+                </CCol>
+                <CCol sm={3}>
+                  <h2>{app.auth().currentUser.displayName}</h2>
+                  <h4>{app.auth().currentUser.email.split("@")[0]}</h4>
+                </CCol>
+                <CCol sm={4}>
+                  <CRow>
+                    <CCol sm={12}>
+                      <Select
+                        id="tag-follow"
+                        name="tag-follow"
+                        options={Object.keys(allTags).map(tagId => {
+                          return { label: allTags[tagId].name, value: tagId };
+                        })}
+                        onChange={selected => setSelectedTagId(selected.value)}
+                      />
+                    </CCol>
+                  </CRow>
+                  <CRow>
+                    <CCol sm={12}>
+                      {followedTagIds.map(
+                        (tagId, idx) =>
+                          allTags[tagId] && (
+                            <CBadge color="success" key={idx} style={{ marginLeft: 3 }}>
+                              {allTags[tagId].name}
+                            </CBadge>
```

Comment:
It would be cool if clicking a badge in the followed list could unfollow it (instead of having a separate button) and perhaps trigger a refresh. Just makes the UI a bit more intuitive

Reply:
Done.